src/test/java/ai/elimu/web/SignOnControllerWeb3Test.java

```diff
@@ -0,0 +1,84 @@
+package ai.elimu.web;
+
+import static org.junit.Assert.*;
+import org.junit.Before;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(locations={
+    "file:src/main/webapp/WEB-INF/spring/applicationContext.xml",
+    "file:src/main/webapp/WEB-INF/spring/applicationContext-jpa.xml"
+})
+public class SignOnControllerWeb3Test {
+    
+    @Autowired
+    private SignOnControllerWeb3 signOnControllerWeb3;
+    
+    private MockMvc mockMvc;
+    
+    @Before
+    public void setup() {
+        assertNotNull(signOnControllerWeb3);
+        mockMvc = MockMvcBuilders.standaloneSetup(signOnControllerWeb3).build();
+        assertNotNull(mockMvc);
+    }
+    
+    @Test
+    public void testHandleGetRequest() throws Exception {
+        RequestBuilder requestBuilder = MockMvcRequestBuilders
+                .get("/sign-on/web3");
+        MvcResult mvcResult = mockMvc.perform(requestBuilder).andReturn();
```

Comment:
We can perform assertion of returned View name like `assertEquals("sign-on-web3", mvcResult.getModelAndView().getViewName());` to check that correct view file was rendered to User

Reply:
@vrudas I was looking for this, but couldn't find it. So thank you for helping me! :-)

Added changes: https://github.com/elimu-ai/webapp/pull/1587/commits/38f1cf9f8382abe25c79084daaa220e7cd6ee612